client/lib/push-notifications/service-worker.js

```diff
@@ -21,3 +33,45 @@ self.addEventListener( 'push', function( event ) {
 		} )
 	);
 } );
+
+self.addEventListener( 'notificationclick', function( event ) {
+	event.notification.close();
+
+	event.waitUntil(
```

Comment:
Mostly basing on the example [here](https://developer.mozilla.org/en-US/docs/Web/API/NotificationEvent). I believe it's because the browser only gives you a certain amount of time to handle the event. `waitUntil` extends the event until the `Promise` returned by `matchAll` is resolved. If you didn't call `waitUntil`, I believe it'd consider the event handled immediately.